spring-cloud-stream/src/main/java/org/springframework/cloud/stream/binder/DefaultBinderFactory.java

```diff
@@ -110,19 +113,30 @@ else if (binderConfigurations.size() == 1) {
 				defaultDomain += ".";
 			}
 			args.add("--spring.jmx.default-domain=" + defaultDomain + "binder." + configurationName);
-			args.add("--spring.application.admin.enabled=false");
 			SpringApplicationBuilder springApplicationBuilder =
 					new SpringApplicationBuilder()
 							.sources(binderConfiguration.getBinderType().getConfigurationClasses())
 							.bannerMode(Mode.OFF)
 							.web(false);
+			// If the environment is not customized and a main context is available, we will set the latter as parent.
+			// This ensures that the defaults and user-defined customizations (e.g. custom connection factory beans)
+			// are propagated to the binder context. If the environment is customized, then the binder context should
+			// not inherit any beans from the parent
+			if (binderProperties.isEmpty() && context != null) {
+				springApplicationBuilder.parent(context);
+			}
+			else if (environment != null && binderConfiguration.isInheritEnvironment()) {
+				StandardEnvironment binderEnvironment = new StandardEnvironment();
+				environment.merge(environment);
```

Comment:
This looks like a typo. I assume the intent is to merge into the `binderEnvironment`?